VideoLocker/src/main/java/org/edx/mobile/view/adapters/CourseSequentialAdapter.java

```diff
@@ -0,0 +1,124 @@
+package org.edx.mobile.view.adapters;
+
+import android.content.Context;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.SectionIndexer;
+import android.widget.TextView;
+
+import org.edx.mobile.R;
+import org.edx.mobile.model.IComponent;
+import org.edx.mobile.model.ISequential;
+import org.edx.mobile.model.IUnit;
+import org.edx.mobile.model.IVertical;
+import org.edx.mobile.view.custom.PinnedSectionListView;
+
+import java.util.List;
+
+/**
+ * Used for pinned behavior.
+ */
+public abstract  class CourseSequentialAdapter extends CourseBaseAdapter
+    implements PinnedSectionListView.PinnedSectionListAdapter, SectionIndexer {
+
+
+    public CourseSequentialAdapter(Context context) {
+        super(context);
+    }
+
+    /**
+     * component can be null.
+     * @IComponent component should be ICourse
+     */
+    public void setData(IComponent component){
+        if ( !(component instanceof ISequential ) )
+            return;//
+        this.rootComponent = component;
+        mData.clear();
+        this.sections = new SectionRow[0];
+        if ( rootComponent != null ) {
+            ISequential course = (ISequential)rootComponent;
+            int sectionsNumber = course.getVerticals().size();
+            sections = new SectionRow[sectionsNumber];
+
+            int sectionPosition = 0, listPosition = 0;
+            for (char i=0; i<sectionsNumber; i++) {
+                IVertical vertical = course.getVerticals().get(i);
+                SectionRow section = new SectionRow(SectionRow.SECTION, vertical );
+                section.sectionPosition = sectionPosition;
+                section.listPosition = listPosition++;
+                sections[sectionPosition] = section;
+                mData.add(section);
+
+                List<IUnit> units = vertical.getUnits();
+                for (int j=0;j<units.size();j++) {
+                    SectionRow item = new SectionRow(SectionRow.ITEM, units.get(j) );
+                    item.sectionPosition = sectionPosition;
+                    item.listPosition = listPosition++;
+                    mData.add(item);
+                }
+
+                sectionPosition++;
```

Comment:
Isn't this value just the same as ``i``? Maybe it would be clearer to make ``sectionPosition`` the loop variable?